config/yaml.go

```diff
@@ -329,36 +325,34 @@ func unmarshalYAMLValue(reader io.ReadCloser, value interface{}) error {
 //
 // TODO: what if someone wanted a literal ${FOO} in config? need a small escape hatch
 func interpolateEnvVars(data []byte) ([]byte, error) {
-	var errs []string
-	data = _interpolationRegex.ReplaceAllFunc(data, func(in []byte) []byte {
-		// remove ${ and }, just look inside the braces
-		valAndMaybeDefault := in[2 : len(in)-1]
-
-		// ${VALUE:DEFAULT} -> VALUE, DEFAULT
-		// ${VALUE} -> VALUE, empty default
-		split := bytes.Split(valAndMaybeDefault, _envSeparator)
-		key := string(split[0])
-
-		if envVal, ok := os.LookupEnv(key); ok {
-			return []byte(envVal)
+	// Is this conversion ok?
+	str := string(data)
+	errs := []string{}
+
+	str = os.Expand(str, func(in string) string {
+		sep := strings.Index(in, _envSeparator)
+
+		var key string
+		var def string
+
+		if sep == -1 {
+			// separator missing - everything is the key ${KEY}
+			key = in
+		} else {
+			// ${KEY:DEFAULT}
+			key = in[:sep]
+			def = in[sep+1:]
```

Comment:
this index logic can easily replace the `split/join` on the other diff, and seems much cleaner.